lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -958,6 +958,46 @@ ::mlir::LogicalResult mlir::tt::ttir::ConvolutionOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// Pooling helper functions
+//===----------------------------------------------------------------------===//
+
+// Check if a AvgPool2dOp or MaxPool2dOp operation is identity
+// Identity operation folding: kernel=[1,1], stride=[1,1], dilation=[1,1],
+// padding=[0,0,0,0]
+template <typename Pool2dOp>
+static bool isIdentityPool2d(Pool2dOp op) {
+  auto kernel = ttmlir::utils::getPairOfInteger<int32_t>(op.getKernel());
+  auto stride = ttmlir::utils::getPairOfInteger<int32_t>(op.getStride());
+  auto dilation = ttmlir::utils::getPairOfInteger<int32_t>(op.getDilation());
+  auto padding = ttmlir::utils::getQuadrupleOfInteger<int32_t>(op.getPadding());
+
+  auto tupleToArray = [](const auto& t) {
```

Comment:
My inner C++ meta programmer got the best of me here, happy to just index in if that's preferred.